### DISCUSSION TITLE
How to add cross project dependency in Sqitch snowflake project

### DISCUSSION THREAD

**Discussion by hrkachhia** [theory/sqitch#709] [Q&A]
Hello,

I wants to manage dependencies between two different snowflake database. For that I have created 2 Sqitch project. Let's  say project SQITCH_DEV for database SQITCH_DEV and project SQITCH_DEV_2 for database SQITCH_DEV_2.  Both project successfully initiated.  I have created test schema  and user table in project SQITCH_DEV. I am successfully deployed project SQITCH_DEV.  I have added table in project SQITCH_DEV_2 with dependency. 

```
/docker-sqitch/sqitch add employee --requires SQITCH_DEV:test  -n 'creates table to track our emloyee.'  
```

The above command  successfully added employee.sql script. 

When I deploy project SQITCH_DEV_2 I am getting below error: 

```
Missing required change: SQITCH_DEV:appschema
```

I got reference from this link : https://contourline.wordpress.com/2016/03/16/an-example-of-using-sqitch-with-cross-project-dependencies/

**Comment by theory** [comment]
Dependencies only work in a single database. The registry Sqitch uses lives in each database, and they're completely independent. If you deploy `SQITCH_DEV` and `SQITCH_DEV_2` to the same database it will work (assuming you deploy `SQITCH_DEV` before ` SQITCH_DEV_2`).

**Reply by hrkachhia** [reply]
Thanks for replied. 

As per your answer and my understanding, Sqitch manage dependencies only single database and multiple schemas for the same database. We can not use Sqitch for cross dependency. Right ?  Please correct if I am wrong.

**Comment by theory** [comment]
That is correct. I'm curious, though: what kinds of dependencies are you looking to enforce _across_ databases?

FWIW on MySQL, you can do it across databases, because in MySQL a database == a schema.

**Reply by hrkachhia** [reply]
I am looking for the cross database dependencies. like in database SQITCH_DEV successfully deployed schema A then and then SQITCH_DEV2 will able to deploy is schema. Something like that.

**Reply by theory** [reply]
But does code in SQITCH_DEV2 directly depend on code in SQITCH_DEV?

**Reply by hrkachhia** [reply]
Yes, It is directly depend on SQITCH_DEV. 
I have followed this tutorial https://contourline.wordpress.com/2016/03/16/an-example-of-using-sqitch-with-cross-project-dependencies/  and based on that I am trying created the example with cross dependency.

**Reply by theory** [reply]
What database platform are you using? If MySQL, you can use the same registry database for different databases on the same server,  but. not different servers. For Postgres and some others, how is SQITCH_DEV2 connecting to and using SQITCH_DEV?
